.github/workflows/autotools.yml

```diff
@@ -11,6 +11,8 @@ jobs:
 
     runs-on: ubuntu-latest
     name: Ubuntu
+    env:
+      CXXFLAGS: "-Werror"
```

Comment:
Should this not also set `CFLAGS` or, alternatively, perhaps only `CPPFLAGS`. I am not 100% sure how to address both C and C++, but most of our files are `C`. Some discussion here:

https://stackoverflow.com/questions/5541946/cflags-ccflags-cxxflags-what-exactly-do-these-variables-control

Reply:
@RossBencina Right, if there is a C code, we should enable the flag there as well. `CPPFLAGS` would be the wrong flags, as CPP stands for C Pre Processor. I've implemented your suggestion using `CFLAGS` and `CXXFLAGS`.